### PR TITLE
fix: batch voting mobile view

### PR DESCRIPTION
**Pull request by VAIBHAV-25**
## Release Notes

- **New Features**
  - Enhanced mobile styling for the Batch Voting component.
  - Conditional rendering of the footer based on user navigation and device type.
  - Added login and signup popups for improved user authentication flow.
  - Introduced a new component for displaying detailed post information in a modal.
  - Real-time feedback on the number of proposals in the cart.
  - New icons and dynamic imports to enhance user interface feedback.
  - Added optional properties to various components for improved functionality.
  - Alert component now displays the user's login address in the Vote Cart.

- **Bug Fixes**
  - Improved error handling for undefined properties in various components.

- **Chores**
  - Streamlined component imports and removed unused code for cleaner architecture.

### CODE REVIEW

File: src/components/BatchVoting/DefaultOptions/OptionWrapper.tsx [hunk 1]
```diff
@@ -137,6 +132,10 @@ const OptionWrapper = ({ className, referendumId, proposalType, forSpecificPost
 		});
 	};
 
+	if (id === null) {
```
Comment (KanishkaRajputd) [inline src/components/BatchVoting/DefaultOptions/OptionWrapper.tsx:135]: isNaN(id)

File: src/components/Home/LatestActivity/BatchVotingWebView.tsx [hunk 2]
```diff
@@ -21,12 +27,28 @@ const BatchVotingWebView = () => {
 				<Button
```
Comment (KanishkaRajputd) [inline src/components/Home/LatestActivity/BatchVotingWebView.tsx:27]: it shoould be a link

File: src/components/TinderStyleVoting/PostInfoComponents/ReferendumV2CardInfo.tsx [hunk 3]
```diff
@@ -34,6 +34,7 @@ interface IReferendumV2CardInfoProps {
 	updateTally?: boolean;
 	post?: any;
 	hideInfo?: boolean;
+	isUsedInBatchVoting?: boolean;
 	isUsedInTinderWebView?: boolean;
```
[the remark this answers is not in the document]
Comment (KanishkaRajputd) [reply src/components/TinderStyleVoting/PostInfoComponents/ReferendumV2CardInfo.tsx:38]: @VAIBHAV-25  why two variables with same product

File: src/components/TinderStyleVoting/VoteCart/index.tsx [hunk 4]
```diff
@@ -156,6 +158,22 @@ const VoteCart: React.FC = () => {
 	return (
 		<section>
 			<article className='px-2'>
+				{loginAddress && (
```
Comment (KanishkaRajputd) [inline src/components/TinderStyleVoting/VoteCart/index.tsx:161]: !!loginAddress?.length &&

File: src/components/TinderStyleVoting/index.tsx [hunk 5]
```diff
@@ -32,13 +37,13 @@ const VotingCards = () => {
 	const currentIndexRef = useRef(currentIndex);
 	const [tempActiveProposals, setTempActiveProposals] = useState([]);
 	const [skippedProposals, setSkippedProposals] = useState<number[]>([]);
+	const [swipeDirection, setSwipeDirection] = useState<string | null>(null);
 
 	const childRefs: any = useMemo(
 		() =>
```
Comment (KanishkaRajputd) [inline src/components/TinderStyleVoting/index.tsx:43]: what's that?

**Review by KanishkaRajputd — CHANGES_REQUESTED** [review]

File: src/components/TinderStyleVoting/index.tsx (hunk 5 above)
Comment (VAIBHAV-25) [reply src/components/TinderStyleVoting/index.tsx:43]: there is a overlay card, which had green, red and yellow color as per card swipe. so for that this state is being used

File: src/components/Home/LatestActivity/BatchVotingWebView.tsx (hunk 2 above)
Comment (VAIBHAV-25) [reply src/components/Home/LatestActivity/BatchVotingWebView.tsx:27]: there is a button and on click of button it should either open loginModal or will redirect

**Review by KanishkaRajputd — APPROVED** [review]

**Pull request merged**